core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala

```diff
@@ -164,14 +164,19 @@ trait ContainerUtils {
             val buffer = java.nio.ByteBuffer.allocate(remain)
             while (remain > 0) {
                 val read = channel.read(buffer)
-                if (read > 0)
-                    remain = read - read.toInt
+                if (read >= 0) {
+                    remain -= read
+                } else {
+                    // read < 0 means premature end of file => terminate loop
+                    remain = 0
+                }
```

Comment:
Per personal discussion: We'll need a test that checks for this condition (and shows "wrong" behavior before the change)